metrics_test.go

```diff
@@ -0,0 +1,326 @@
+package metrics
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/transcovo/go-chpr-logger"
+	"github.com/transcovo/go-chpr-metrics/utils/tests"
+)
+
+var port int = 12345
+var localhost = "127.0.0.1"
+
+/*
+Ensure unicity in port number (tests are ran in parallel)
+*/
+func getUniquePort() string {
+	port++
+	return strconv.Itoa(port)
+}
+
+/*
+Start an UDP server that does nothing. Waits for the test to finish to close the server
+Notify the channel when cleaning is done
+*/
+func simpleUDPServer(host, port string, end chan string) {
+	pc, err := net.ListenPacket("udp", host+":"+port)
+	if err != nil {
+		fmt.Println("Error creating the udp server", err)
+		panic(err)
+	}
+	end <- ""  // Notify that ready
```

Comment:
well actually, I prefer to keep it that way, this would be misleading. I changed the "" to an explicit message for the reader to ease his work.